lib/saml/settings_service.rb

```diff
@@ -31,9 +34,9 @@ def settings
     end
 
     ## Makes an external web call to get IDP metadata and populates SETTINGS
-    def fetch_idp_metadata
+    def create_settings_from_metadata
       parser = OneLogin::RubySaml::IdpMetadataParser.new
-      parser.parse_remote(SAML_CONFIG['metadata_url'], true, settings: settings)
+      parser.parse(self.class.metadata, settings: settings)
```

Comment:
@omgitsbillryan, I just pushed a commit that makes it validate the SSL stuff when querying for the metadata. The Net::HTTP API is not the prettiest, but I believe this'll address your concern.